### PR TITLE
chore: clean up login command messaging structure

### PR DESCRIPTION
**Pull request by mshaaban0**
## Summary

- Clean up and structure logging messages for `login` command
- Migrate `utils/async.js` to typescript

## Screenshots (if appropriate):
Screenshot 2022-09-06 at 15 08 14

Screenshot 2022-09-06 at 00 43 52

### CODE REVIEW

File: lib/cmds/login.ts [hunk 1]
```diff
@@ -55,28 +54,25 @@ export const login = async ({
     token = managementTokenFlag
   } else {
     if (managementToken) {
-      log()
-      log(
-        `Looks like you already stored a management token on your system. ${chalk.dim(
-          `(Located at ${await getConfigPath()})`
-        )}`
+      console.log(
+        `You're already logged in! to logout type: ${chalk.green(
```
Comment (denkristoffer) [inline lib/cmds/login.ts:58]:
```suggestion
        `You're already logged in! To logout type: ${chalk.green(
```

File: lib/cmds/login.ts [hunk 2]
```diff
@@ -55,28 +54,25 @@ export const login = async ({
     token = managementTokenFlag
   } else {
     if (managementToken) {
-      log()
-      log(
-        `Looks like you already stored a management token on your system. ${chalk.dim(
-          `(Located at ${await getConfigPath()})`
-        )}`
+      console.log(
```
Comment (denkristoffer) [inline lib/cmds/login.ts:57]: Why did we drop the `log` utility?

File: lib/cmds/login.ts (hunk 1 above)
Comment (denkristoffer) [inline lib/cmds/login.ts:58]: Personally I prefer the old message with the "maybe you want to log out?" coming later, I think it's a good separation whereas the logout hint is a bit harder to spot here.

File: lib/cmds/login.ts (hunk 2 above)
Comment (massao) [reply lib/cmds/login.ts:57]: The `log` utility is just a wrapper on the `console.log`.
The others utility have a bit more of logic/styling involved, but we could keep it as log for consistency 🤔 
https://github.com/contentful/contentful-cli/blob/master/lib/utils/log.js

Comment (mshaaban0) [reply lib/cmds/login.ts:59]: Yup, it's just a wrapper that confuses me with other utility loggers we have, overtime I have to click to see what it is, I'd say it's not worth it. Maybe in another PR I'll just do a quick find and replace

Comment (denkristoffer) [reply lib/cmds/login.ts:59]: I think that's okay but let's kill it everywhere then, yeah.

**Review by denkristoffer — APPROVED** [review]

**Pull request merged**